File: speechtotext/chirp-nodejs/chirp.js
// Import the Cloud Speech-to-Text library
const speech = require('@google-cloud/speech').v2;
const fs = require('fs');

// Instantiates a client
const client = new speech.SpeechClient({apiEndpoint: "us-central1-speech.googleapis.com"});

// Your local audio file to transcribe
const audioFilePath = "gs://<<bucket-name>>/audio-files/output.wav";
// Full recognizer resource name
const recognizerName = "projects/<<project-number>>/locations/us-central1/recognizers/<<recognizer>>";
// The output path of the transcription result.
const workspace = "gs://<<bucket-name>>/transcripts";

const recognitionConfig = {
  autoDecodingConfig: {},
  model: "chirp",
  features: {
  enableWordTimeOffsets: true,
  },
  languageCode: 'en-IN'
};

const audioFiles = [
  { uri: audioFilePath }
];
const outputPath = {
  gcsOutputConfig: {
    uri: workspace
  }
};

async function transcribeSpeech() {
  const transcriptionRequest = {
    recognizer: recognizerName,
    config: recognitionConfig,
    files: audioFiles,
    recognitionOutputConfig: outputPath,
  };

const [operation] = await client.batchRecognize(transcriptionRequest);
    const [response] = await operation.promise();
    console.log(response);
}

transcribeSpeech();

File: speechtotext/chirp-python/chirp.py
import os

from google.api_core.client_options import ClientOptions
from google.cloud.speech_v2 import SpeechClient
from google.cloud.speech_v2.types import cloud_speech

# Instantiates a client
#client = SpeechClient()

# Instantiates a client
client = SpeechClient(
    client_options=ClientOptions(
        api_endpoint="us-central1-speech.googleapis.com",
    )
)

# The output path of the transcription result.
workspace = "gs://<<bucket-name>>/transcripts"

# The name of the audio file to transcribe:
gcs_uri = "gs://<<bucket-name>>/audio-files/output.wav"

# Recognizer resource name:
name = "projects/<<project-number>>/locations/us-central1/recognizers/<<recognizername>>"

config = cloud_speech.RecognitionConfig(
  auto_decoding_config={},
  model="chirp",
  features=cloud_speech.RecognitionFeatures(
  enable_word_time_offsets=True,
  ),
)

output_config = cloud_speech.RecognitionOutputConfig(
  gcs_output_config=cloud_speech.GcsOutputConfig(
    uri=workspace),
)

files = [cloud_speech.BatchRecognizeFileMetadata(
    uri=gcs_uri
)]

request = cloud_speech.BatchRecognizeRequest(
    recognizer=name, config=config, files=files, recognition_output_config=output_config
)
operation = client.batch_recognize(request=request)
print(operation.result())
